base/setup/lib/utils/partlist.c

```diff
@@ -882,146 +1002,70 @@ AddPartitionToDisk(
-                    /*
-                     * Dismount the partition since RawFS owns it, and set its
-                     * format to unknown (may or may not be actually formatted).
-                     */
-                    DismountVolume(PartEntry);
-                    PartEntry->FormatState = UnknownFormat;
-                    PartEntry->FileSystem[0] = L'\0';
-                }
-            }
-            else
-            {
-                PartEntry->FormatState = Preformatted;
-            }
-        }
-        else
+        /* Attach and mount the volume */
+        Status = MountVolume(&Volume->Info, PartEntry->PartitionType);
+        if (!NT_SUCCESS(Status))
         {
-            PartEntry->FormatState = UnknownFormat;
+            DPRINT1("Failed to mount volume '%S', Status 0x%08lx\n",
+                    Volume->Info.DeviceName, Status);
         }
 
-        /* Retrieve the partition volume label */
-        if (PartitionHandle)
-        {
-            Status = NtQueryVolumeInformationFile(PartitionHandle,
-                                                  &IoStatusBlock,
-                                                  &LabelBuffer,
-                                                  sizeof(LabelBuffer),
-                                                  FileFsVolumeInformation);
-            if (NT_SUCCESS(Status))
-            {
-                /* Copy the (possibly truncated) volume label and NULL-terminate it */
-                RtlStringCbCopyNW(PartEntry->VolumeLabel, sizeof(PartEntry->VolumeLabel),
-                                  LabelInfo->VolumeLabel, LabelInfo->VolumeLabelLength);
-            }
-            else
-            {
-                DPRINT1("NtQueryVolumeInformationFile() failed, Status 0x%08lx\n", Status);
-            }
-        }
-
-        /* Close the partition */
-        if (PartitionHandle)
-            NtClose(PartitionHandle);
+        //
+        // FIXME: TEMP Backward-compatibility: Set the FormatState
+        // flag in accordance with the FileSystem volume value.
+        //
+        /*
+         * MountVolume() determines whether the given volume is actually
+         * unformatted, if it was mounted with RawFS and the partition
+         * type has specific values for FAT volumes. If so, the volume
+         * stays mounted with RawFS (the FileSystem is "RAW"). However,
+         * if the partition type has different values, the volume is
+         * considered as having an unknown format (it may or may not be
+         * formatted) and the FileSystem value has been emptied.
+         */
+        if (IsUnknown(&Volume->Info)) // (!*...Info.FileSystem)
+            Volume->FormatState = UnknownFormat;
+        else if (IsUnformatted(&Volume->Info)) // (_wcsicmp(...Info.FileSystem, L"RAW") == 0)
+            Volume->FormatState = Unformatted;
+        else // !IsUnknown && !IsUnformatted == IsFormatted
+            Volume->FormatState = Formatted;
```

Comment:
The 3 slash-slash comments above do not seem to really tell us the purpose if the if tests.

Reply:
hm yes indeed, I added these in order to remember what the `IsUnknown`, `IsUnformatted` .... macros are doing, basically. I will consider removing these.